tee-worker/litentry/core/stf-task/receiver/src/handler/assertion.rs

```diff
@@ -0,0 +1,167 @@
+use itp_types::OpaqueCall;
+use lc_stf_task_sender::AssertionBuildRequest;
+use litentry_primitives::{Assertion, Identity, Web2Network};
+use log::error;
+use parachain_core_primitives::VCMPError;
+use std::{format, string::String, sync::Arc};
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::chrono::{offset::Utc as TzUtc, TimeZone};
+
+#[cfg(feature = "std")]
+use chrono::{offset::Utc as TzUtc, TimeZone};
+
+pub(crate) struct AssertionHandler<
+	K: ShieldingCryptoDecrypt + ShieldingCryptoEncrypt + Clone,
+	O: EnclaveOnChainOCallApi,
+	C: CreateExtrinsics,
+	M: AccessNodeMetadata,
+	A: AuthorApi<Hash, Hash>,
+	S: StfEnclaveSigning,
+	H: HandleState,
+> {
+	pub(crate) req: AssertionBuildRequest,
+	pub(crate) context: Arc<StfTaskContext<K, O, C, M, A, S, H>>,
+}
+
+impl<K, O, C, M, A, S, H> TaskHandler for AssertionHandler<K, O, C, M, A, S, H>
+where
+	K: ShieldingCryptoDecrypt + ShieldingCryptoEncrypt + Clone,
+	O: EnclaveOnChainOCallApi,
+	C: CreateExtrinsics,
+	M: AccessNodeMetadata,
+	M::MetadataType: IMPCallIndexes + VCMPCallIndexes,
+	A: AuthorApi<Hash, Hash>,
+	S: StfEnclaveSigning,
+	H: HandleState,
+	H::StateT: SgxExternalitiesTrait,
+{
+	type Error = VCMPError;
+	type Result = ();
+
+	fn on_process(&self) -> Result<Self::Result, Self::Error> {
+		match self.req.assertion.clone() {
+			Assertion::A1 => lc_assertion_build::a1::build(self.req.vec_identity.clone()),
+
+			Assertion::A2(guild_id, handler) => {
+				for identity in &self.req.vec_identity {
+					if let Identity::Web2 { network, .. } = identity {
+						if matches!(network, Web2Network::Discord) {
+							if let Err(e) =
+								lc_assertion_build::a2::build(guild_id.clone(), handler.clone())
+							{
+								error!("error verify assertion2: {:?}", e)
+							} else {
+								// When result is Ok,
+								break
+							}
+						}
+					}
+				}
+				Err(VCMPError::Assertion2Failed)
+			},
+
+			Assertion::A3(guild_id, handler) => {
+				for identity in &self.req.vec_identity {
+					if let Identity::Web2 { network, .. } = identity {
+						if matches!(network, Web2Network::Discord) {
+							if let Err(e) =
+								lc_assertion_build::a3::build(guild_id.clone(), handler.clone())
+							{
+								error!("error verify assertion2: {:?}", e)
+							} else {
+								// When result is Ok,
+								break
+							}
+						}
+					}
+				}
+				Err(VCMPError::Assertion3Failed)
+			},
+
+			Assertion::A4(mini_balance, from_date) => {
+				let mini_balance: f64 = (mini_balance / (10 ^ 12)) as f64;
+				lc_assertion_build::a4::build(
+					self.req.vec_identity.clone(),
+					String::from_utf8(from_date.into_inner()).unwrap(),
+					mini_balance,
+				)
+			},
+
+			Assertion::A5(twitter_account, original_tweet_id) => lc_assertion_build::a5::build(
+				self.req.vec_identity.to_vec(),
+				twitter_account,
+				original_tweet_id,
+			),
+			Assertion::A6 => lc_assertion_build::a6::build(self.req.vec_identity.to_vec()),
+
+			Assertion::A7(mini_balance, year) => {
```

Comment:
It's strange `mini_balance` is still in the code, which I remember I had renamed it into `min_balance`. Can we simply rename it now?